### DISCUSSION TITLE
Linking Course Faculty short hand with actual full name

### DISCUSSION THREAD

**Discussion by RyanBarclay** [VikeLabs/courseup#198] [Ideas]
an example is PHYS 223 if you look for "PHYS 323" in winter 2020 it shows up, but "physics 223" does not. Would assume you would just have to link the short hand and full written out name instead of just looking for "physics" in description or what ever is being scraped

**Comment by aomi** [comment]
Does the UVic one even do this? I've personally never searched using a _longer_ form of any of the subjects. This isn't hard to fix (moreso add a feature using synonyms)

**Reply by FlyteWizard** [reply]
The UVic course search does not, but it does let you type the course name (i.e. Algorithms and Data Structures). (Which CourseUp also allows). I think it might be worth adding. (Especially for new students).

**Reply by aomi** [reply]
This isn't difficult to add but I'm just not sure if it would add value as adding synonyms (ie PHYS = PHYSICS) would make entries with the word "physics" less prominent in the results. Just a thought.

**Reply by RyanBarclay** [reply]
I had two of my roommates try and look for classes using the faculty name instead of the short hand. Just wanted to relay that by doing this not all courses under the faculty show up.

**Reply by aomi** [reply]
"Physics" isn't a faculty and falls under "Faculty of Science" which isn't something we do. 

I feel there's some wrong terminology being thrown around. I'd just like to clarify the title. 
> Using full form of a subject (ie. Physics = PHYS) does not return results

might be better. Adding faculty would be adding a "faculty" field on all course records and is completely unrelated to this afaik.

**Reply by aomi** [reply]
Since it's kinda related, here's some of the data that shows "no results".
![image](https://user-images.githubusercontent.com/1757744/121767044-cb8c1980-cb0a-11eb-98da-c269a7216805.png)
There is a pattern with users insert non-space delimited subject + code. 

I also see the case like you mention where someone has searched the full subject (`psychology 365`).

**Reply by aomi** [reply]
I've tuned the search to return compound search queries (ie allowing `PHYS111` instead of `PHYS 111`) and for experimenting, I did add `PHYS = PHYSICS` (case insensitive). @RyanBarclay Try it out and tell me if that's what you had in mind :+1:

**Reply by RyanBarclay** [reply]
LGTM!
